core/controllers/admin_test.py

```diff
@@ -707,6 +709,93 @@ def test_upload_topic_similarities(self):
 
         self.logout()
 
+    def test_update_feature_flag_rules(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+
+        feature = platform_parameter_registry.Registry.create_feature_flag(
+            'test_feature_1', 'feature for test.', 'dev')
+        new_rule_dicts = [
+            {
+                'filters': [
+                    {
+                        'type': 'server_mode',
+                        'conditions': [['=', 'dev']]
+                    }
+                ],
+                'value_when_matched': True
+            }
+        ]
+
+        feature_list_ctx = self.swap(
+            platform_feature_services, 'ALL_FEATURES_LIST', [feature.name])
+        feature_set_ctx = self.swap(
+            platform_feature_services, 'ALL_FEATURES_NAMES_SET',
+            set([feature.name]))
+        with feature_list_ctx, feature_set_ctx:
+            response_dict = self.get_json('/adminhandler')
```

Comment:
This should be in a separate test (e.g. ``test_get_handler_includes_all_feature_flags`` or similar). Prefer keeping tests specific to testing certain behaviors to avoid tests unexpectedly passing/failing due to unrelated issues (e.g. I wouldn't expect a test corresponding to updating feature flags to necessarily fail if the getter has an issue).